ycmd.el

```diff
@@ -701,6 +721,42 @@ Returns the new value of `ycmd-force-semantic-completion'."
       (setq it (list it)))
     (mapcar 'expand-file-name it)))
 
+(defvar ycmd--mode-keywords-loaded nil)
+
+(defun ycmd--get-keywords (buffer)
+  "Get syntax keywords for BUFFER."
+  (with-current-buffer buffer
+    (let ((mode major-mode))
+      (unless (memq mode ycmd--mode-keywords-loaded)
```

Comment:
We only need to send the keywords once to the server, therefore the check. It's the same in the vim YCM plugin.